articles/i2p-docker-services/GithubPagesMirror.md

```diff
@@ -0,0 +1,233 @@
+desirable, for instance for i2p applications that collaborate primarily by way
+of Github, but who wish to have a presence on i2p as well. Probably other cases
+where long-term pseudonymity is desirable.
+
+So, just to be clear
+--------------------
+
+  * *For this tutorial to be unlinkable to your real identity, your github*
+   *must be unlinkable to your real identity*.
+  * *This is pseudonymous, location-obfuscated mirroring of a github pages*
+   *site.*
+
+Now with that out of the way, let's get down to mirroring a Github Pages site,
+for example this exact wiki, to your own i2p eepSite.
+
+Preparation
+-----------
+
+To get started, you'll need an i2pd docker container with an http server tunnel
+prepared. For convenience's sake, you can just use the one from the previous
+tutorial. First, make sure you've created your network and specified a subnet,
+as in the first tutorial:
+
+```
+docker network create --subnet 172.81.81.0/29 eepsite
+```
+
+And alter the tunnels.conf in your tunnels.conf to look like this:
+
+```
+[DARKHTTPD]
+type = http
+host = eepsite-darkhttpd
+port = 8080
+inbound.length = 3
+outbound.length = 3
+keys = darkhttpd.dat
+
+[GITHUB]
+type = http
+host = eepsite-github
+port = 8090
+inbound.length = 3
+outbound.length = 3
+keys = github.dat
+```
+
+Once you've done that, you can "docker build" the new container and your
+original, darkhttpd tunnel will continue to exist at the same destination but a
+new unique destination will be generated for your forthcoming Github mirror.
+From the perspective of someone browsing the i2p network, and the other routers
+in the i2p network, there will be no good reason to believe that either of those
+tunnels are running on the same i2p router. If you no longer want the original
+darkhttpd tunnel, then you can just delete the [DARKHTTPD] section from your
+tunnels.conf.
+
+Obfuscating the IP address of your container host
+-------------------------------------------------
+
+In order to fetch your repositories from git pseudonymously, or in order to
+obfuscate your location as a git user, you will need to ensure that git is
+configured to fetch clearnet resources via Tor. In order to do this, we'll add a
+container running Tor to our docker network, with an exposed SocksPort.
+Additionally, we will do no actual building or generation of the Github Pages
+mirror until the containers are running and present on the network. In order to
+safely host any site that requests resources from the clearnet, you will need to
+do a similar procedure to route it through Tor. For git, this procedure is
+simple to do so it makes a good example.
+
+For the purposes of this demonstration, we're going to assume that the pertinent
+adversaries do not have sufficient power to link your fetching of packages via
+the rubygems package manager to your eepsite activity. If that's an issue for
+you, I kinda hope you didn't need this tutorial. It also assumes that you are
```

Comment:
"didn't" should probably be "don't" as the user is currently reading the tutorial, therefore it is present tense :)